lib/streamlit/elements/widgets/selectbox.py

```diff
@@ -128,7 +214,8 @@ def selectbox(
         placeholder: str = "Choose an option",
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
-    ) -> T | None:
+        accept_new_options: Literal[False, True] | bool = False,
```

Comment:
Why not just `bool` as type?

Reply:
because of mypy 😞 If I am not mistaken, it did not allow me to overload the function for the different return types based on the boolean default value. So for the overloads I had to use the `Literal` variant. But only using `Literal[False, True]` then here in the function leads to false complains because, apparently, passing `True | False` directly is different to passing a variable which is `True | False`.

Reply:
yeah that seems to work! I could swear that I have seen mypy complain, but maybe I did it the other way around and deleted the `bool` part.